File: CTCI/Ch1Arrays/Py/isUnique.py
"""
This checks if a word contains
all unique letters.
"""


def isUnique(word):
    wordList = []
    for letter in word:
        if letter in wordList:
            return False
        else:
            wordList.append(letter)
    return True


print(isUnique("turtle"))
print(isUnique("Libre"))

File: CTCI/Ch1Arrays/ts/is-unique.ts
function isUnique(str: string): Boolean {
  let sety = new Set();

  for (let s of str) {
    if (sety.has(s)) {
      return false;
    }

    sety.add(s);
  }
  return true;
}

console.log(isUnique("12344"));
console.log(isUnique("1234"));
